Dockerfile.develop

```diff
@@ -90,16 +90,23 @@ RUN true\
 # won't invalidate earlier download layers
 RUN go mod download
 
-# Install controller-gen
+# Export the Go binary path for controller-gen and ginkgo CLIs
+ENV PATH $HOME/go/bin:$PATH
+
+# Install controller-gen to generate util code and Kubernetes YAMLs for API changes
+RUN true \
+    && go install sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_GEN_VERSION} \
```

Comment:
Error before this: 

`which: no controller-gen in (/root/go/bin/:/usr/local/go/bin:/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/kubebuilder/bin:)`

```
$ make run fmt

Makefile:206: warning: overriding commands for target `fmt'
Makefile:139: warning: ignoring old commands for target `fmt'
./scripts/build_devimage.sh "docker"
Pulling dev image kserve/modelmesh-controller-develop:8ac28e213eb25d80...
...
./scripts/develop.sh make fmt
which: no controller-gen in (/root/go/bin/:/usr/local/go/bin:/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/kubebuilder/bin:)
./scripts/fmt.sh || (echo "Linter failed: $?"; git status; exit 1)
golangci-lint............................................................
...
```

Reply:
So the `controller-gen` CLI was not installed properly since we upgraded to Go 1.18 due to the [`go get` behavior change](https://go.dev/doc/go-get-install-deprecation)

I also updated the controller-gen version to the last one aligned with Go 1.19. to get address this [error](https://github.com/kubernetes-sigs/controller-tools/issues/613):

```
$ make run generate

panic: interface conversion: ast.Expr is *ast.CompositeLit, not *ast.Ident

goroutine 1212 [running]:
sigs.k8s.io/controller-tools/pkg/loader.(*referenceCollector).Visit(0xc0074b0300, {0xd7f0e8?, 0xc003882240})
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.4.0/pkg/loader/refs.go:136 +0x105
go/ast.Walk({0xd7ca80?, 0xc0074b0300?}, {0xd7f0e8?, 0xc003882240?})
        /usr/lib/golang/src/go/ast/walk.go:51 +0x62
go/ast.walkExprList({0xd7ca80, 0xc0074b0300}, {0xc0052bc000?, 0x1, 0xc07260?})
        /usr/lib/golang/src/go/ast/walk.go:26 +0x91
...
```

Even after that upgrade we still need to address this (and possibly more) error:

```
$ make run generate

/workspace/apis/serving/v1alpha1/servingruntime_webhook.go:34:2: invalid field type: interface{RESTMapper() k8s.io/apimachinery/pkg/api/meta.RESTMapper; Scheme() *k8s.io/apimachinery/pkg/runtime.Scheme; sigs.k8s.io/controller-runtime/pkg/client.Reader; sigs.k8s.io/controller-runtime/pkg/client.Writer; sigs.k8s.io/controller-runtime/pkg/client.StatusClient; sigs.k8s.io/controller-runtime/pkg/client.SubResourceClientConstructor}
Error: not all generators ran successfully

```

Reply:
https://github.com/kserve/modelmesh-serving/issues/387